src/resources/lib/modules/connman.py

```diff
@@ -1044,12 +1049,36 @@ def dbus_error_handler(self, error):
             self.oe.set_busy(0)
             err_name = error.get_dbus_name()
             if 'InProgress' in err_name:
-                self.disconnect_network()
+                if self.net_disconnected != 1:
+                    self.disconnect_network()
+                else
```

Comment:
Should be `else:`

Reply:
thanks, must have missed that, will fix it now, sorry this PR is messy i trying to get all the wifi fixes together here though.